amethyst_gltf/src/format/importer.rs

```diff
@@ -0,0 +1,328 @@
+            Ok(parse_data_uri(uri)?)
+        } else {
+            let path = base_path.parent().unwrap_or(Path::new("./")).join(uri);
+            Ok(read_to_end(source.clone(), &path)?)
+        };
+        let data = data_res?;
+
+        if data.len() < buffer.length() {
+            let path = json::Path::new().field("buffers").index(index);
+            return Err(Error::BufferLength(path));
+        }
+        buffers.push(data);
+    }
+    Ok(buffers)
+}
+
+fn validate_standard(unvalidated: gltf::Unvalidated) -> Result<Gltf, Error> {
+    Ok(unvalidated.validate_completely()?)
+}
+
+fn validate_binary(unvalidated: gltf::Unvalidated, has_bin: bool) -> Result<Gltf, Error> {
+    use gltf::json::validation::Error as Reason;
+
+    let mut errs = vec![];
+    {
+        let json = unvalidated.as_json();
+        for (index, buffer) in json.buffers.iter().enumerate() {
+            let path = || json::Path::new().field("buffers").index(index).field("uri");
+            match index {
+                0 if has_bin => if buffer.uri.is_some() {
+                    errs.push((path(), Reason::Missing));
+                },
+                _ if buffer.uri.is_none() => {
+                    errs.push((path(), Reason::Missing));
+                }
+                _ => {}
+            }
+        }
+    }
+
+    if errs.is_empty() {
+        Ok(unvalidated.validate_completely()?)
+    } else {
+        Err(Error::Validation(errs))
+    }
+}
+
+fn import_standard<'a>(
+    data: &'a [u8],
+    source: Arc<AssetSource>,
+    base_path: &Path,
+) -> Result<(Gltf, Buffers), Error> {
+    let unvalidated = Gltf::from_slice(data)?;
+    let gltf = validate_standard(unvalidated)?;
+    let bin = None;
+    let mut buffers = Buffers(vec![]);
+    for buffer in load_external_buffers(source, base_path, &gltf, bin)? {
+        buffers.0.push(buffer);
+    }
+    Ok((gltf, buffers))
+}
+
+fn import_binary<'a>(
+    data: &'a [u8],
+    source: Arc<AssetSource>,
+    base_path: &Path,
+) -> Result<(Gltf, Buffers), Error> {
+    let gltf::Glb {
+        header: _,
+        json,
+        bin,
+    } = gltf::Glb::from_slice(data)?;
+    let unvalidated = Gltf::from_slice(json)?;
+    let bin = bin.map(|x| x.to_vec());
+    let gltf = validate_binary(unvalidated, bin.is_some())?;
+    let mut buffers = Buffers(vec![]);
+    for buffer in load_external_buffers(source, base_path, &gltf, bin)? {
+        buffers.0.push(buffer);
+    }
+    Ok((gltf, buffers))
+}
+
+pub fn get_image_data(
+    image: &gltf::Image,
+    buffers: &Buffers,
+    source: Arc<AssetSource>,
+    base_path: &Path,
+) -> Result<(Vec<u8>, ImageFormat), Error> {
+    match image.data() {
+        gltf::image::Data::View { view, mime_type } => {
+            let data = buffers.view(&view).unwrap();
+            let format = match mime_type {
+                "image/png" => ImageFormat::Png,
+                "image/jpeg" => ImageFormat::Jpeg,
+                _ => unreachable!(),
```

Comment:
Technically, this could also be e.g. `.bmp`

Reply:
No support in gltf got other than jpg and bmp

Reply:
Jpg and png I mean ofc

Reply:
Is that defined by reference or the Rust implementation?

Reply:
The glTF specification allows only JPEG and PNG.